lib/engine/game/base.rb

```diff
@@ -1118,6 +1118,11 @@ def all_potential_upgrades(tile)
           .select { |t| upgrades_to?(tile, t) }
           .reject(&:blocks_lay)
       end
+
+      # In all_potential_upgrades context
+      def from_tile_manifest?(tile)
+        !tile.hex || tile.hex.name == 'A1'
```

Comment:
this is not good

Reply:
Have made a new attempt

Reply:
what i didn't like was the hex.name == 'A1' why is that needed is there another way?